### PR TITLE
Add auxiliary model with validator for type-specific usage instructions

### PR DESCRIPTION
**Pull request by ml-evs**
See discussion in https://github.com/marda-alliance/metadata_extractors_schema/issues/49 for why we can't do this directly in LinkML.

### CODE REVIEW

File: marda_registry/models/__init__.py [hunk 1]
```diff
@@ -3,9 +3,52 @@
...
+        """
+
+        if not v:
+            return v
+
+        supported_filetypes = values.get("supported_filetypes")
+        usage_to_fts = {}
+
+        bad_fts = set()
+        if usage := v:
```
Comment (PeterKraus) [inline marda_registry/models/__init__.py:33]: You could just do:
```
if not v:
    return v
else:
    usage = v
```
and avoid the fancy `:=` operator that I had to google. It's not like evaluating `v` is expensive...

File: marda_registry/models/__init__.py [hunk 2]
```diff
@@ -3,9 +3,52 @@
...
+                        usage_to_fts[t] = instruction
+                        if t not in supported_filetypes:
+                            bad_fts.add((t, "Bad filetype {t} found in instruction"))
+
+        for t in supported_filetypes:
+            if t not in usage_to_fts:
+                bad_fts.add((t, f"No instruction for {t}"))
+
+        if bad_fts:
+            raise ValueError("Unsupported filetypes: " + "\n".join(bad_fts))
```
Comment (PeterKraus) [inline marda_registry/models/__init__.py:49]: Maybe raise a `pydantic.ValidationError` instead, just in case someone downstream would be catching those?

**Review by PeterKraus — CHANGES_REQUESTED** [review]

File: marda_registry/models/__init__.py (hunk 2 above)
Comment (ml-evs) [reply marda_registry/models/__init__.py:48]: Raising `ValueError` from validators is actually intentional, pydantic converts it to `ValidationError` automatically

Comment (ml-evs) [reply marda_registry/models/__init__.py:48]: (see e.g. https://docs.pydantic.dev/latest/concepts/validators/#field-validators -- ctrl+f for ValueError as I can't link to the exact paragraph)

**Pull request merged**